### PR TITLE
Update GDASApp (gdas.cd) hash

### LINKED ISSUE #2222: Update GDASApp hash
### What new functionality do you need?

GDASApp `develop` now accesses JEDI libraries via submodules.  This issue is opened to update the GDASApp hash cloned by global-workflow `develop`.

### What are the requirements for the new functionality?

GDASApp uses submodules for JEDI repositories.

### Acceptance Criteria

GDASApp successfully builds from global-workflow.

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
Update GDASApp (`gdas.cd`) hash to clone JEDI libraries from submodules 

# Description
GDASApp `develop` at [`639ef16`](https://github.com/NOAA-EMC/GDASApp/commit/639ef1673b8aca1b2631153cfdc89c134c08ea88) clones JEDI repositories using submodules.  This PR updates the GDASApp (`gdas.cd`) hash which the global-workflow references to [`639ef16`](https://github.com/NOAA-EMC/GDASApp/commit/639ef1673b8aca1b2631153cfdc89c134c08ea88).  

Since GDASApp `develop` at [`639ef16`](https://github.com/NOAA-EMC/GDASApp/commit/639ef1673b8aca1b2631153cfdc89c134c08ea88) builds and runs on Hercules, logic in `sorc/build_all.sh` is modified to include Hercules as a viable machine on which to build GDASApp from the global-workflow.

Global-workflow based GDASApp ctests identified two additional issues.  
1. `jobs/JGDAS_GLOBAL_OCEAN_ANALYSIS_BMAT` references `ocnanalrun` when it should reference `ocnanalbmat`
2. `parm/config/gfs/config.resources` specified two threads for `ocnanalrun` when one thread should be used

These issues are resolved in this PR.

One last detail regards the Hera logic for the GDASApp build in `sorc/build_all.sh`   Script `build_all.sh`  executes the following to identify the machine on which the build is being executed
```#------------------------------------
# GET MACHINE
#------------------------------------
export COMPILER="intel"
source gfs_utils.fd/ush/detect_machine.sh
```
Script `detect_machine.sh` appends `COMPILER` to the machine name on Hera and Cheyenne
```
 # Append compiler (only on machines that have multiple compilers)
if [ $MACHINE_ID = hera ] || [ $MACHINE_ID = cheyenne ]; then
    MACHINE_ID=${MACHINE_ID}.${COMPILER}
fi
```
Thus, `build_all.sh` sets `MACHINE_ID` to `hera.intel` when executed on Hera.  The existing logic in `build_all.sh`
```
# Optional DA builds
if [[ "${_build_ufsda}" == "YES" ]]; then
   if [[ "${MACHINE_ID}" != "orion" && "${MACHINE_ID}" != "hera" ]]; then
      echo "NOTE: The GDAS App is not supported on ${MACHINE_ID}.  Disabling build."
   else
```
fails to build GDASApp because `hera.intel` does not equal `hera`.  This PR changes the Hera logic to
```
# Optional DA builds
if [[ "${_build_ufsda}" == "YES" ]]; then
   if [[ "${MACHINE_ID}" != "orion" && "${MACHINE_ID}" != "hera.intel" && "${MACHINE_ID}" != "hercules" ]]; then
      echo "NOTE: The GDAS App is not supported on ${MACHINE_ID}.  Disabling build."
   else
```
GDASApp does not yet have a modulefile for gnu so `hera.intel` is sufficient.

Resolves #2222 

# Type of change
- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)?  **NO**
- Does this change require a documentation update? **NO**

# How has this been tested?
- Clone and build on Hera, Hercules, and Orion

GDASApp g-w based ctests run.  All tests pass except `test_gdasapp_soca_JGDAS_GLOBAL_OCEAN_ANALYSIS_VRFY`.  A GDASApp fix for this ctest is under development.

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] New and existing tests pass with my changes

### CODE REVIEW

File: sorc/build_all.sh [hunk 1]
```diff
@@ -129,7 +129,7 @@ build_opts["ww3prepost"]="${_verbose_opt} ${_build_ufs_opt}"
 
 # Optional DA builds
 if [[ "${_build_ufsda}" == "YES" ]]; then
-   if [[ "${MACHINE_ID}" != "orion" && "${MACHINE_ID}" != "hera" ]]; then
+   if [[ "${MACHINE_ID}" != "orion" && "${MACHINE_ID}" != "hera.intel" && "${MACHINE_ID}" != "hercules" ]]; then
```
Comment (aerorahul) [inline sorc/build_all.sh:132]: @RussTreadon-NOAA 
The UFSWM has removed the `COMPILER` add-on to the `MACHINE_ID` and we will be removing it from `gfs-utils` as well soon.
The GSI, and other places will also be updated soon. 
FYI.

Comment (RussTreadon-NOAA) [reply sorc/build_all.sh:132]: Glad to hear this.  It's odd to have `MACHINE_ID` represent a machine + compiler.

**Review by aerorahul — APPROVED** [review]
LGTM.

**Comment by RussTreadon-NOAA** [comment]
Thank you @aerorahul for acting so fast on this PR!

**Pull request merged**